File: admin_app/lib/api/entity/OrderEntity.dart
import 'package:admin_app/api/entity/CityEntity.dart';
import 'package:admin_app/api/entity/PropertiesEntity.dart';
import 'package:admin_app/api/entity/UserEntity.dart';
import 'package:json_annotation/json_annotation.dart';

part 'OrderEntity.g.dart';

@JsonSerializable()
class OrderEntity {
  int id;
  UserEntity creator;
  UserEntity? specialist;
  CityEntity city;
  AddressEntity addressTo;
  AddressEntity addressFrom;
  double price;
  bool customPrice;
  bool outCity;
  bool active;
  DateTime startDate;
  DateTime endDate;
  DateTime createDate;

  OrderEntity({required this.id, required this.creator, this.specialist, required this.city, required this.addressTo, required this.addressFrom, required this.price, required this.customPrice, required this.outCity, required this.startDate, required this.active, required this.endDate, required this.createDate});

  factory OrderEntity.fromJson(Map<String, dynamic> json) => _$OrderEntityFromJson(json);
  Map<String, dynamic> toJson() => _$OrderEntityToJson(this);
}

File: server/GNext/src/main/java/com/thedeveloper/gnext/entity/OrderEntity.java
package com.thedeveloper.gnext.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;

@Entity
@Table(name = "orders")
@Data
public class OrderEntity {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    Long id;
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "creator", nullable = true, referencedColumnName = "phone")
    UserEntity creator;

    @ManyToOne(fetch = FetchType.LAZY, optional = true)
    @JoinColumn(name = "specialist", nullable = true, referencedColumnName = "phone")
    UserEntity specialist;
    @ManyToOne
    @JoinColumn(name = "city", nullable = true,referencedColumnName = "name")
    CityEntity city;
    @ManyToOne
    @JoinColumn(name = "addressTo", nullable = true,referencedColumnName = "id")
    AddressEntity addressTo;
    @ManyToOne
    @JoinColumn(name = "addressFrom", nullable = true,referencedColumnName = "id")
    AddressEntity addressFrom;
    boolean active = true;
    double price;
    String description;
    boolean customPrice;
    boolean outCity;
    Date startDate;
    Date endDate;
    Date createDate;
    String file;

}
